src/GraphQL/DI/DefaultServiceProvider.cs

```diff
@@ -11,15 +11,22 @@ namespace GraphQL
     public sealed class DefaultServiceProvider : IServiceProvider
     {
         /// <summary>
-        /// Gets an instance of the specified type. Can not return <see langword="null"/> but may throw exception.
+        /// Gets an instance of the specified type. Returns <see langword="null"/> for interfaces.
+        /// Can not return <see langword="null"/> for classes but may throw exception.
         /// </summary>
         /// <param name="serviceType">Desired type</param>
         /// <returns>An instance of <paramref name="serviceType"/>.</returns>
-        public object GetService(Type serviceType)
+        public object? GetService(Type serviceType)
         {
             if (serviceType == null)
                 throw new ArgumentNullException(nameof(serviceType));
 
+            if (serviceType == typeof(IServiceProvider))
```

Comment:
`if (serviceType == typeof(IServiceProvider) || serviceType == typeof(DefaultServiceProvider))` to be precise.

Reply:
Or perhaps if it can be assigned to this.GetType in case of inheritance.

Reply:
Nevermind; class is sealed.

Reply:
Would you add a check?

Reply:
https://github.com/graphql-dotnet/graphql-dotnet/pull/2769